types/node/index.d.ts

```diff
@@ -93,10 +93,17 @@ interface NodeRequireFunction {
 interface NodeRequire extends NodeRequireFunction {
     resolve(id: string): string;
     cache: any;
-    extensions: any;
+    extensions: NodeExtensions;
     main: NodeModule | undefined;
 }
 
+interface NodeExtensions {
+  '.js': (m: NodeModule, filename: string) => any;
```

Comment:
These all have the same type of value, shouldn't it just be the string indexer then?

Reply:
I think it's better to leave them for documentation purposes. From type checker point of view it doesn't matter, but for human reader it should make it more clear what is expected usage.